core/analysis/pipeline_token_stream.cpp

```diff
@@ -0,0 +1,295 @@
+              "pipeline_token_stream from jSON arguments: %s",
+              PROPERTIES_PARAM_NAME.c_str(), PIPELINE_PARAM_NAME.c_str(), args.c_str());
+            return false;
+          }
+        } else {
+          IR_FRMT_ERROR(
+            "Failed to read '%s' member as object while constructing "
+            "pipeline_token_stream from jSON arguments: %s",
+            PIPELINE_PARAM_NAME.c_str(), args.c_str());
+          return false;
+        }
+      }
+    } else {
+      IR_FRMT_ERROR(
+        "Failed to read '%s' attribute as array while constructing "
+        "pipeline_token_stream from jSON arguments: %s",
+        PIPELINE_PARAM_NAME.c_str(), args.c_str());
+      return false;
+    }
+  } else {
+    IR_FRMT_ERROR(
+      "Not found parameter '%s' while constructing pipeline_token_stream, "
+      "arguments: %s",
+      PIPELINE_PARAM_NAME.c_str(),
+      args.c_str());
+    return false;
+  }
+  if (options.pipeline.empty()) {
+    IR_FRMT_ERROR(
+      "Empty pipeline found while constructing pipeline_token_stream, "
+      "arguments: %s", args.c_str());
+    return false;
+  }
+  return true;
+}
+
+bool normalize_json_config(const irs::string_ref& args, std::string& definition) {
+  options_normalize_t options;
+  if (parse_json_config(args, options)) {
+    definition.clear();
+    definition.append("{\"").append(PIPELINE_PARAM_NAME).append("\":[");
+    bool first{ true };
+    for (auto analyzer : options.pipeline) {
+      if (first) {
+        first = false;
+      } else {
+        definition.append(",");
+      }
+      definition.append("{\"").append(TYPE_PARAM_NAME)
+        .append("\":\"").append(analyzer.first).append("\",")
+        .append("\"").append(PROPERTIES_PARAM_NAME).append("\":")
+        .append(analyzer.second).append("}");
+    }
+    definition.append("]}");
+    return true;
+  }
+  return false;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief args is a jSON encoded object with the following attributes:
+/// pipeline: Array of objects containing analyzers definition inside pipeline.
+/// Each definition is an object with the following attributes:
+/// type: analyzer type name (one of registered analyzers type)
+/// properties: object with properties for corresponding analyzer
+////////////////////////////////////////////////////////////////////////////////
+irs::analysis::analyzer::ptr make_json(const irs::string_ref& args) {
+  irs::analysis::pipeline_token_stream::options_t options;
+  if (parse_json_config(args, options)) {
+    return std::make_shared<irs::analysis::pipeline_token_stream>(options);
+  } else {
+    return nullptr;
+  }
+}
+
+
+REGISTER_ANALYZER_JSON(irs::analysis::pipeline_token_stream, make_json,
+  normalize_json_config);
+
+NS_END
+
+NS_ROOT
+NS_BEGIN(analysis)
+
+pipeline_token_stream::pipeline_token_stream(const pipeline_token_stream::options_t& options)
```

Comment:
`pipeline_token_stream` must accept `std::vector<irs::analysis::analyzer::ptr>&&` to ensure that pipeline analyzers **owns** sub-analyzers, otherwise a one analyzer could be used in different pipelines which is error prone.